monitoring/openshift/cronjob-registry-prune.yaml

```diff
@@ -0,0 +1,150 @@
+#    value: "*/2 * * * *"
+    required: true
+  - name: "JOB_SERVICE_ACCOUNT"
+    displayName: "Service Account Name"
+    description: "Name of the Service Account To Exeucte the Job As."
+    value: "registry-pruner"
+    required: true
+  - name: "SUCCESS_JOBS_HISTORY_LIMIT"
+    displayName: "Successful Job History Limit"
+    description: "The number of successful jobs that will be retained"
+    value: "5"
+    required: true
+  - name: "FAILED_JOBS_HISTORY_LIMIT"
+    displayName: "Failed Job History Limit"
+    description: "The number of failed jobs that will be retained"
+    value: "5"
+    required: true
+  - name: "JOB_BACKOFF_LIMIT"
+    displayName: "Job Backoff Limit"
+    description: "The number of attempts to try for a successful job outcome (default: 6)"
+    value: "1"
+    required: false
+objects:
+- kind: ConfigMap
+  apiVersion: v1
+  metadata:
+    name: ${JOB_SOURCE_SCRIPT}
+    namespace: ${NAMESPACE}
+    labels:
+      template: "cronjob-registry-prune"
+  data:
+    ${JOB_SOURCE_SCRIPT}: |
+      NOOP=echo
+      echo "running: oc adm prune images --keep-younger-than=96h"; date
+      oc adm prune images --keep-younger-than=96h
+      rc=$?; if [ $rc -ne 0 ] ; then
+        date
+        echo "sleeping... zzz"
+        sleep 60s
+        echo "Dry-Run Prune failed, aborting..."
+        exit $rc;
+      fi
+      ${NOOP} oc adm prune images --keep-younger-than=96h
+      rc=$?; if [ $rc -ne 0 ] ; then
+        date
+        echo "sleeping... zzz"
+        sleep 60s
+        echo "Prune failed, aborting..."
+        exit $rc;
+      fi
+      ${NOOP} oc rollout latest dc/docker-registry -n default
+      ${NOOP} oc rollout status dc/docker-registry -n default
+- kind: "CronJob"
+  apiVersion: "batch/v1beta1"
+  metadata:
+    name: "${JOB_NAME}"
+    namespace: ${NAMESPACE}
+    labels:
+      template: "cronjob-registry-prune"
+  spec:
+    schedule: "${SCHEDULE}"
+    concurrencyPolicy: "Forbid"
+    successfulJobsHistoryLimit: "${{SUCCESS_JOBS_HISTORY_LIMIT}}"
+    failedJobsHistoryLimit: "${{FAILED_JOBS_HISTORY_LIMIT}}"
+    jobTemplate:
+      spec:
+        backoffLimit: ${JOB_BACKOFF_LIMIT}
+        template:
+          spec:
+            containers:
+              - name: "${JOB_NAME}"
+                image: "openshift/jenkins-slave-base-centos7"
+                command:
+                  - "/bin/bash"
+                  - "-c"
+                  - "source /etc/config/${JOB_SOURCE_SCRIPT} || :"
+                volumeMounts:
+                  - mountPath: "/etc/config"
+                    name: "registry-prune-volume"
+            volumes:
+              - configMap:
+                  items:
+                    - key: ${JOB_SOURCE_SCRIPT}
+                      path: ${JOB_SOURCE_SCRIPT}
+                  name: ${JOB_SOURCE_SCRIPT}
+                name: registry-prune-volume
+            restartPolicy: "Never"
+            terminationGracePeriodSeconds: 30
+            activeDeadlineSeconds: 500
```

Comment:
Is this the time limit for the Job to complete?

Reply:
yes, activeDealineSeconds is the job completion time limit (~8min),  if you have some measurements for job duration that are more appropriate, please feel free to update.

Reply:
Looks like PROD usually finishes in under 5 min
```
# ll /var/log/oc-prune/*.err | awk '{print $8}' | grep ^07 | sort -u
07:00
07:01
07:02
07:03
07:04
```

Reply:
going to bump to 30 minutes (1600s) for now as the job will also stay active to log the status of the docker-registry rollout.